js/ui/calendar/ui.calendar.selection.strategy.js

```diff
@@ -0,0 +1,227 @@
+import { noop } from '../../core/utils/common';
+import Class from '../../core/class';
+import dateUtils from '../../core/utils/date';
+
+const abstract = Class.abstract;
+
+const DAY_INTERVAL = 86400000;
+
+const CalendarSelectionStrategy = Class.inherit({
```

Comment:
Shouldn't we use usual ES6 classes instead of out custom `Class.inherit`?

Reply:
done